templates/common/baremetal/files/NetworkManager-mdns-hostname.yaml

```diff
@@ -0,0 +1,21 @@
+filesystem: "root"
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/40-mdns-hostname"
+contents:
+  inline: |
+    #!/bin/bash
+    STATUS=$2
+    case "$STATUS" in
+        dhcp4-change|dhcp6-change|hostname)
+        logger -s "NM mdns-hostname triggered by ${2}."
+        set +e
+        t_hostname=$(hostname)
+        if [ -z "${t_hostname}" ]; then
+           t_hostname="localhost"
```

Comment:
`localhost.localdomain` ?

Reply:
Running 'hostname' from shell returns just the hostname without a domain name(e.g: localhost), while GOLANG os.Hostname() returns host+domain-name.
The baremetal-runtimecfg relevant code (utils.ShortHostname [1]) should work OK for both cases.
[1] https://github.com/openshift/baremetal-runtimecfg/blob/master/pkg/utils/utils.go#L22